### ISSUE TITLE
DOI not minting all the way

### ISSUE THREAD

**Issue by janhohenheim** [dandi/dandi-archive#1715]
### Bug description
I uploaded a dandiset and received an ID, but no DOI link: [
Dataset of human medial temporal lobe neurons, scalp and intracranial EEG during a verbal working memory task ](https://dandiarchive.org/dandiset/000574).
The ID is DANDI:000574/0.231010.1809. If I try to manually create a DOI link from this, I end up with https://doi.org/10.48324/dandi.000574/0.231010.1809. This DOI is not found, however.

#### Expected behaviour
The DOI link should be generated and be valid.

#### Actual behaviour
Only the ID is generated, but not the DOI link. Additionally, the DOI link I suspect should be generated is vacant.

### How to reproduce
Unclear

### Your personal set up
I used the current DANDI website to publish my draft

- <details><summary>Logs</summary>

</details>

**Comment by mvandenburgh** [comment]
Thanks for the bug report @janhohenheim. I reran the DOI generation internally on your dandiset and it succeeded, so you'll see a DOI link on https://dandiarchive.org/dandiset/000574 now. 

I'll look into why this is happening more generally.

**Comment by mvandenburgh** [comment]
It looks like this dandiset was published on October 10th, which predated the release of the fix (#1703) of this bug. So, I'll go ahead and close this issue.

As for how this went undetected - after merging that bug fix, I did do a one off query to detect and fix any published dandisets that were missing DOIs, but it seems I missed some. This particular dandiset had an empty string for a DOI; I believe I queried for `NULL` DOIs when doing that repairing work, so I must have missed this one.
